application/src/main/java/com/github/pieter_groenendijk/service/AccountService.java

```diff
@@ -0,0 +1,25 @@
+package com.github.pieter_groenendijk.service;
+
+import java.util.Date;
+import java.util.UUID;
+import com.github.pieter_groenendijk.model.Account;
+import com.github.pieter_groenendijk.repository.AccountRepository;
+import com.github.pieter_groenendijk.repository.IAccountRepository;
+
+public class AccountService {
+
+    private IAccountRepository _accountRepository;
+
+    public AccountService() {
+        _accountRepository = new AccountRepository();
```

Comment:
Niet gemakkelijk testbaar zonder DI

Reply:
Hoe zou je je dit dan voorstellen?

Reply:
Dependency Injection, dus:

```java
public AccountService(IAccountRepository repository) {
    this.accountRepository = repository
}
```
Je wilt namelijk de service apart van de repo willen kunnen testen, dus bij het testen kan dan een stub (of hoe dat ook officieel heet) gebruikt worden i.p.v. de echte.

Reply:
De Service wordt aangemaakt vanuit de controller. Moet je daar dan al een repository aanmaken om hem door te kunnen geven aan deze constructor?

Reply:
De aanroep van deze constructor zit in de controller. Moet ik daar dan al een constructor aanmaken om deze mee te kunnen geven?

Reply:
Over het algemeen krijg je idd een omgedraaid systeem. In pseudocode

```java
repository = new Repository
service = new Service(repository)
controller = new Controller(service)
```

Het voordeel hiervan is dat wanneer je het wilt gaat unit testen, en dus niet de daadwerkelijke database wilt gebruiken, je een mock van de interface `IAccountRepository` kan meegeven. Zo kan je deterministisch (en snel) een waarde verwachten van de repository functies. Je kan dus code testen onafhankelijk van de database.

Reply:
Een ander voordeel hiervan is dat je dezelfde service instantie bijv. kan gebruiken bij meerdere controllers.

Reply:
Kijk maar even of het zo is zoals je het bedoeld heb, dan mag deze ook geresolved worden.